app/components/shared/cluster/ClusterServicesOverview.tsx

```diff
@@ -1,20 +1,40 @@
-// @ts-nocheck File not migrated fully to TS
 import _ from 'lodash';
-import PropTypes from 'prop-types';
 import React from 'react';
 import { useSelector } from 'react-redux';
 import i18n from 'i18next';
 import { Link } from 'react-router-dom';
+import type { FunctionComponent, ReactNode } from 'react';
+
 import { ErrorMessage, LoadingOverlay, Message, Table } from '../../basic';
 import ClusterService from './ClusterService';
-import { clusterServiceBgColor, clusterServiceEnum, clusterServiceStatusEnum, clusterServiceStatuses } from './consts';
+import { clusterServiceBgColor, clusterServiceEnum, clusterServiceStatusEnum } from './consts';
 import './ClusterServicesOverview.css';
 import { createPagesMap } from '../../../actions/pageMenu';
+import type { ClusterService as ClusterServiceName, ClusterServices, ClusterServiceStatus } from './types';
+import type { ReduxState } from '../../../reducers';
 
-export default function ClusterServicesOverview({ services, clickable, isFetching, fetchingError, header }) {
+interface ClusterServicesOverviewProps {
+    services: ClusterServices;
+    clickable?: boolean;
+    isFetching?: boolean;
+    fetchingError?: string;
+    header?: ReactNode;
+}
+const ClusterServicesOverview: FunctionComponent<ClusterServicesOverviewProps> = ({
+    services = _.mapValues(clusterServiceEnum, () => ({
+        status: clusterServiceStatusEnum.Unknown as ClusterServiceStatus,
+        is_external: false
+    })),
+    clickable = false,
+    isFetching = false,
+    fetchingError = '',
+    header = null
```

Comment:
Looking at the the use cases of those props, it seems like they don't need default value assignment 🎉 

```suggestion
    clickable,
    isFetching,
    fetchingError,
    header
```

Reply:
Let me understand it better. Do you mean that e.g. if we don't have `clickable` passed it will be `undefined` and in the place where it is used L61:
```
{clickable && isSystemHealthPagePresent ? (
```
it will work the same as when `clickable` === `false`?

If so, then right, we can have it also `undefined` and it'll work, but are we losing anything by being a bit more descriptive here? 🤔

Reply:
Yeah, that was one of the reasons behind the idea 😁 

I don't think that there are some outstanding benefits which we could get due to romoving those default props assignments
But as they're not necessary (in that particular case) I thought that ommiting them would simplify code a little 🎉 

But other than that, I don't think that there is much of an additional benefit to it  😅 
Because of that, I'm leaving a decision up to you and as it seems to be a not required change, I'm approving this PR 🎉

Reply:
Thanks for clarification. In that case, I'll leave it as is and keep the original functionality.